client/client.go

```diff
@@ -34,6 +34,50 @@ import (
 	"github.com/canonical/pebble/internals/wsutil"
 )
 
+type Requester interface {
+	// Do performs the HTTP transaction using the provided options.
+	Do(ctx context.Context, opts *RequestOptions) (*RequestResponse, error)
+
+	// Transport returns the HTTP transport in use by the underlying HTTP client.
+	Transport() *http.Transport
+}
+
+type RequestType int
+
+const (
+	RawRequest RequestType = iota
+	SyncRequest
+	AsyncRequest
+)
+
+type RequestOptions struct {
+	Type    RequestType
+	Method  string
+	Path    string
+	Query   url.Values
+	Headers map[string]string
+	Body    io.Reader
+}
+
+type RequestResponse struct {
+	StatusCode int
+	ChangeID   string
```

Comment:
Should we add a comment to the `ChangeID` field, something like "this is normally only set for async requests: it's the change ID of the asynchronous operation started".

Similar for `Result`, I think it needs a comment about how it's just the inner payload part (however you want to say that clearly).